frontends/web/src/routes/account/send/send.tsx

```diff
@@ -270,30 +271,36 @@ class Send extends Component<Props, State> {
       amountError: undefined,
       feeError: undefined,
     });
-    if (this.sendDisabled()) {
+    const txInput = this.getValidTxInputData();
+    if (!txInput) {
       return;
     }
-    const txInput = this.txInput();
     if (this.proposeTimeout) {
       clearTimeout(this.proposeTimeout);
       this.proposeTimeout = null;
     }
     this.setState({ isUpdatingProposal: true });
-    this.proposeTimeout = setTimeout(() => {
-      const propose = apiPost('account/' + this.getAccount()!.code + '/tx-proposal', txInput)
-        .then(result => {
-          const pos = this.pendingProposals.indexOf(propose);
-          if (this.pendingProposals.length - 1 === pos) {
-            this.txProposal(updateFiat, result);
-          }
-          this.pendingProposals.splice(pos, 1);
-        })
-        .catch(() => {
-          this.setState({ valid: false });
-          this.pendingProposals.splice(this.pendingProposals.indexOf(propose), 1);
-        });
-      this.pendingProposals.push(propose);
-    }, 400);
+    // defer the transaction proposal
+    this.proposeTimeout = setTimeout(async () => {
+      const proposePromise = accountApi.proposeTx(this.getAccount()!.code, txInput);
+      // keep this as the last known proposal
+      this.lastProposal = proposePromise;
+      try {
+        const result = await proposePromise;
+        // continue only if this is the most recent proposal
+        if (proposePromise === this.lastProposal) {
+          this.txProposal(updateFiat, result);
+        }
```

Comment:
Always been like this, so no probs. Just for my understanding: would it also be ok to await `proposePromise` inside this `if` stmt?

Reply:
I haven't tested, but the purpose here is to do this check when we get a result. This is in the edge case that if the first promise takes longer than the latest promise. We don't want the promise that is the slowest. But we want the result of the promise that was called last.

does that make sense?